serviceAccessConfig/tests/unit/test_base.py

```diff
@@ -0,0 +1,194 @@
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import ConfigParser
+import pytest
+import os
+import sys
+
+from mock import patch
+
+# Our test utilities
+import unittest_utils as utils
+
+sys.path.insert(0, utils.get_code_path())
+
+from serviceAccessConfig.accessrulegenerator import ServiceAccessGenerator
+from serviceAccessConfig.generatorexceptions import *
+
+
+# ======================================================================
+def test_allowed_client_ip_addrs():
+    """Test that we get the expected cidr blocks from a valid config
+       file."""
+
+    gen = ServiceAccessGenerator('%s/ip_data.cfg' % utils.get_data_path())
+    cidr_blocks = gen._get_allowed_client_ip_addrs()
+    expected = '8.0.0.0/24,132.168.2.0/8,154.12.0.0/16,18.168.1.1/32'
+
+    assert cidr_blocks == expected
+
+
+# ======================================================================
+@patch('serviceAccessConfig.accessrulegenerator.logging')
+def test_allowed_client_ip_addrs_error_config(mock_logging):
+    """Test that an error is generated when the config file does not
+       exist, i.e. cannot be read"""
+
+    gen = ServiceAccessGenerator(
+        '%s/ip_data_syntax_error.cfg' % utils.get_data_path()
+    )
+    with pytest.raises(ServiceAccessGeneratorConfigError) as excinfo:
+        cidr_blocks = gen._get_allowed_client_ip_addrs()
+
+    assert mock_logging.error.called
+    assert 'tests/data/ip_data_syntax_error.cfg' in str(excinfo.value)
+
+
+# ======================================================================
+@patch('serviceAccessConfig.accessrulegenerator.logging')
+def test_allowed_client_ip_addrs_no_config(mock_logging):
+    """Test that an error is generated when the config file has a syntax
+       error"""
+
+    gen = ServiceAccessGenerator('%s/foo.cfg' % utils.get_data_path())
+    with pytest.raises(ServiceAccessGeneratorConfigError) as excinfo:
+        cidr_blocks = gen._get_allowed_client_ip_addrs()
+
+    assert mock_logging.error.called
+    assert 'tests/data/foo.cfg' in str(excinfo.value)
+
+
+# ======================================================================
+@patch('serviceAccessConfig.accessrulegenerator.os.path.exists',
+       return_value=False)
+@patch('serviceAccessConfig.accessrulegenerator.os')
+@patch('serviceAccessConfig.accessrulegenerator.logging')
+def test_restart_service_systemd(mock_logging, mock_os, mock_ospath):
+    """Test that the proper systemd command is issued to start a service"""
+
+    gen = ServiceAccessGenerator('%s/ip_data.cfg' % utils.get_data_path())
+    gen.service_name = 'foo'
+    with pytest.raises(ServiceAccessGeneratorServiceRestartError) as excinfo:
+        gen._restart_service()
+
+    assert mock_logging.error.called
+    mock_logging.error.assert_called_with('foo restart failed')
+    mock_os.system.assert_called_with('systemctl restart foo.service')
```

Comment:
It would be better to mock out the point of system interaction, instead of trying to run the service, failing, and looking for the error. I'm tempted to add a foo.service to my system just to force a test failure :P